apps/main/src/datastore/server/create/create.ts

```diff
@@ -48,60 +48,38 @@ const getCreateMetadata = (userRef: DocumentReference<DocumentData>) => ({
 });
 
 export const createUser = async (
-  newUserEdits: Pick<User, 'name' | 'avatarUrl'>,
+  user: Pick<User, 'name' | 'email' | 'avatarUrl'>,
   authUser: AuthUser,
 ) => {
-  // Any authorized user can create a record for themselves in the DB; but they
-  // can still only modifiy their own record due to the use of authUser.uid.
   const path = `users/${authUser.uid}`;
-  const newUserRef = await getDocRefInternal(UserSchema, path);
-
-  const newUserDetails = await newUserRef.firestore.runTransaction(async (transaction) => {
-    // Check for invites for the user's email address or UID
-    const invitesByEmailSnapshot = await (
-      await getCollectionRefInternal(InviteSchema, 'invites')
-    )
-      .where('email', '==', authUser.email)
-      .where('status', '==', 'pending')
-      .get();
-    const invitesByUidSnapshot = await (
-      await getCollectionRefInternal(InviteSchema, 'invites')
-    )
-      .where('uid', '==', authUser.uid)
-      .where('status', '==', 'pending')
-      .get();
-
-    const allInvites = [
-      ...invitesByEmailSnapshot.docs,
-      ...invitesByUidSnapshot.docs,
-    ];
-
-    const organizationRefsMap = new Map<
-      string,
-      {id: string, path: string}
-    >();
-
-    allInvites.forEach((doc) => {
-      (doc.data().organizationRefs ?? []).forEach((orgRef) => {
-        organizationRefsMap.set(orgRef.path, orgRef);
-      });
-      transaction.update(doc.ref, { status: 'accepted' });
-    });
-
-    const newUser: User = {
-      ...newUserEdits,
-      id: newUserRef.id,
-      path: asDocPath(newUserRef.path),
-      organizationRefs: Array.from(organizationRefsMap.values()),
-      ...getCreateMetadata(newUserRef),
-    };
+  const userRef = await getDocRefInternal(UserSchema, `users/${authUser.uid}`);
+  const ref = await getDocRefInternal(UserSchema, path);
+  const newUser = {
+    ...user,
+    id: ref.id,
+    path: asDocPath(ref.path),
+    ...getCreateMetadata(userRef),
+  };
+  await ref.set(newUser);
+  await sendVerificationEmail(user.email);
+  return ref.get();
+};
 
-    transaction.set(newUserRef, newUser);
-    return newUser;
+const sendVerificationEmail = async (email: string) => {
+  const app = getFirebaseAdminApp();
+  const auth = getAuth(app);
+  const link = await auth.generateEmailVerificationLink(email);
+  // The firestore-send-email extension is used in this project.
+  // I will add a document to the `mail` collection.
+  const { getFirestore } = await import('firebase-admin/firestore');
+  const db = getFirestore(app);
+  await db.collection('mail').add({
+    to: email,
+    message: {
+      subject: 'Verify your email for Preem Machine',
+      html: `Please verify your email by clicking this link: <a href="${link}">${link}</a>`,
```

Comment:
this code can live in this datastore specific file; but the combined use of this code and the auth library means we need to move this into apps/main/src/auth/server/